### PR TITLE
ACS-3227 Add some more comprehensive REST tests for creating rules.

### PR DESCRIPTION
**Pull request by tpage-alfresco**

### CODE REVIEW

**Comment by tpage-alfresco** [comment]
This PR covers most of the outstanding tests for creating rules (until we add more fields). I have two more PRs to go for this ticket:
1. Omit the rule name (requires a production code change)
2. Further negative tests around GET (particularly permissions tests)

**Comment by tpage-alfresco** [comment]
Branch build failed with:

> [ERROR] testGetPathElements_DocLib(org.alfresco.rest.api.tests.NodeApiTest)  Time elapsed: 2.422 s  <<< FAILURE!
> java.lang.AssertionError: Status code 403 returned, but expected 200
> 	at org.alfresco.rest.api.tests.NodeApiTest.testGetPathElements_DocLib(NodeApiTest.java:658)

**Review by KikoUK — COMMENTED** [review]
Is adding a description to a rule part of the Create endpoint?

**Review by KikoUK — COMMENTED** [review]
Shall we also have a negative e2e for creating a Rule with forbidden symbols in its name, e.g. !@£$%^&*()_+|":}{

Since name field is required, I guess we should have an e2e covering a request with an empty name?

**Comment by montgolfiere** [comment]
> Shall we also have a negative e2e for creating a Rule with forbidden symbols in its name, e.g. !@£$%^&*()_+|":}{

I'm not sure it's critical. Just remember, the rule name is actually mapped to cm:title in the content model (rather than cm:name that has additional constraints).

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/CreateRulesTests.java [hunk 1]
```diff
@@ -79,13 +84,91 @@ public void createRule()
...
+
+    /** Check we can't create a rule under a document node. */
+    @Test (groups = { TestGroup.REST_API, TestGroup.RULES })
+    public void tryToCreateRuleUnderDocument()
+    {
+        STEP("Create a document.");
+        FileModel fileModel = dataContent.usingUser(user).usingSite(site).createContent(getRandomFileModel(TEXT_PLAIN));
+
+        RestRuleModel ruleModel = new RestRuleModel();
+        ruleModel.setName("");
```
Comment (montgolfiere) [inline packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/CreateRulesTests.java:160]: I assume missing name will be covered later, ie. as per PR notes, more changes to come in follow-on PRs:

> Omit the rule name (requires a production code change)

**Review by montgolfiere — APPROVED** [review]

**Comment by tpage-alfresco** [comment]
> Shall we also have a negative e2e for creating a Rule with forbidden symbols in its name, e.g. !@£$%^&*()_+|":}{

Good idea - I'll add that as part of my next PR.

> Since name field is required, I guess we should have an e2e covering a request with an empty name?

See my earlier comment: https://github.com/Alfresco/alfresco-community-repo/pull/1239#issuecomment-1192464663

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/CreateRulesTests.java (hunk 1 above)
Comment (tpage-alfresco) [reply packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/CreateRulesTests.java:160]: This looks like a copy/paste mistake. This test shouldn't have an empty rule name.

**Comment by tpage-alfresco** [comment]
> Is adding a description to a rule part of the Create endpoint?

This will come later along with a lot of other fields.

**Pull request merged**